src/orbits/kepler.jl

```diff
@@ -1,110 +1,230 @@
+const G = PhysicalConstants.CODATA2018.G
+const G_val = ustrip(u"cm^3/g/s^2", G)
+const M₀ = π / 2.0
 
+"""
+    KeplerianOrbit(; kwargs...)
+Keplerian orbit parameterized by the basic observables of a transiting 2-body system.
+# Parameters
+* `a` - The semi-major axis, nominally in AU
+* `aRₛ` - The ratio of the semi-major axis to the star radius. Aliased to `aRs`
+* `b` - The impact parameter, bounded between 0 ≤ b ≤ 1
+* `ecc` - The eccentricity of the closed orbit, bounded between 0 ≤ ecc < 1
+* `P` - The orbital period of the planet, nominally in days
+* `ρₛ` - The spherical star density, nominally in g/cc. Aliased to `rho_s`
+* `Rₛ` - The star mass, nominally in solar radii. Aliased to `R_s`
+* `t₀` - The midpoint time of the reference transit, same units as `P`. Aliased to `t0`
+* `incl` - The inclination of the orbital plane relative to the axis perpendicular to the
+           reference plane, nominally in degrees
+* `Ω` - The longitude of the ascending node, nominally in radians. Aliased to `Omega`
+* `ω` - The argument of periapsis, same units as `Ω`. Aliased to `omega`
+"""
 struct KeplerianOrbit <: AbstractOrbit
     a
+    aRₛ
+    b
     ecc
-    period
-    ρ_star
-    r_star
-    m_star
-    m_planet
-    m_total
+    P
+    ρₛ
+    Rₛ
     n
-    a_star
-    a_planet
-    K0
-    M0
-    t_periastron
-    t0
-    tref
+    t₀
     incl
-    sin_incl
-    cos_incl
+    Ω
+    ω
 end
 
-const G_grav = 2942.2062175044193
-
-function KeplerianOrbit(; period, m_planet=0, r_star=1, m_star=1, ρ_star = 3 * m_star / (4 * π * r_star^3), ecc=nothing, t_periastron=0, t0 = 0 + t_periastron)
-    a = cbrt(G_grav * (m_star + m_planet) * period^2 / (4 * π^2))
-    m_total = m_star + m_planet
-    n = 2 * π / period
-    a_star = a * m_planet / m_total
-    a_planet = -a * m_star / m_total
-    K0 = n * a / m_total
-    if ecc === nothing
-        M0 = 0.5 * π
-        incl_factor = 1
-    else
-        error("not implemented")
-    end
-    incl = 0.5 * π
-    sin_incl, cos_incl = sincos(incl)
-
-    tref = t_periastron - t0
-    return KeplerianOrbit(a,
-                          ecc,
-                          period,
-                          ρ_star,
-                          r_star,
-                          m_star,
-                          m_planet,
-                          m_total,
-                          n,
-                          a_star,
-                          a_planet,
-                          K0,
-                          M0,
-                          t_periastron,
-                          t0,
-                          tref,
-                          incl,
-                          sin_incl,
-                          cos_incl)
+# Enable keyword dispatch and argument name aliasing
+@kwdispatch KeplerianOrbit(;
+    Omega => Ω,
+    omega => ω,
+    aRs => aRₛ,
+    rho_s => ρₛ,
+    aRs => aRₛ,
+    Rs => Rₛ,
+    t0 => t₀,
+)
+
+@kwmethod function KeplerianOrbit(;ρₛ, Rₛ, ecc, P, t₀, incl)
+    Ω = π / 2.0
```

Comment:
this value is defined in a constant above

Reply:
Sure!